test/test_jit_py3.py

```diff
@@ -433,6 +433,32 @@ def foo():
                 return MyCoolNamedTuple(4, 5.5, [3])
             print(foo.graph)
 
+    def test_already_scripted_submodule(self):
+        class MyModel(nn.ModuleDict):
+            def __init__(self):
+                super().__init__()
+                self.layers: List[nn.Module] = []
+                for i in range(3):
+                    layer = torch.jit.script(nn.Linear(10, 10))
+                    self.add_module(str(i), layer)
+                    self.layers.append(layer)
+                self.layers = tuple(self.layers)
+
+            def forward(self, x):
+                for m in self.layers:
+                    x = m.forward(x)
+                return x
+
+        input = torch.rand(10)
+
+        # Set the seed before creating my MyModel instances so that the Parameters for both
+        # are identical and their outputs can be compared.
+        torch.manual_seed(0)
```

Comment:
It's more idiomatic to do `with freeze_rng_state():` so that this test doesn't affect others